brian2/synapses/synapses.py

```diff
@@ -1570,3 +1575,17 @@ def _add_synapses_generator(self, j, n, skip_if_invalid=False, namespace=None, l
                                         check_units=False,
                                         run_namespace=namespace)
         codeobj()
+
+    def _check_synapse_parse(self, parsed, namespace):
```

Comment:
I'll change it to `_check_parsed_synapse_generator`. But I will add a suitable doc string to clarify the function's purpose.